[expo-screen-capture] Clarify and document how it works with the Android privacy changes. (#31865)
expo/expo: docs/pages/versions/unversioned/sdk/screen-capture.mdx

This is especially important on Android since the [`android.media.projection`](https://developer.android.com/about/versions/android-5.0.html#ScreenCapture) API allows third-party apps to perform screen capture or screen sharing (even if the app is in the background).

On Android, the screen capture callback works without additional permissions only for Android 14+. **You don't need to request or check permissions for blocking screen capture or using the callback on Android 14+.**

> If you want to use the screen capture callback on Android 13 or lower, you need to add the `READ_MEDIA_IMAGES` permission to your `AndroidManifest.xml` file.
> You can do that using the `android.permissions` key in your app config. [Learn more.](https://docs.expo.dev/guides/permissions/#android)

Reviewer: ```suggestion
If you want to use the screen capture callback on Android 13 or lower, you need to add the `READ_MEDIA_IMAGES` permission to your `AndroidManifest.xml` file. You can use the `android.permissions` key in your app config. See [Android permissions](/guides/permissions/#android) for more information.
```

Let's use description for the link instead of learn more as per writing style guide.

> > **warning** The `READ_MEDIA_IMAGES` permission can be added only for apps with a need for broad access to photos. [Learn more.](https://support.google.com/googleplay/android-developer/answer/14115180)

Reviewer: ```suggestion
> **warning** The `READ_MEDIA_IMAGES` permission can be added only for apps needing broad access to photos. See [Details on Google Play's Photo and Video Permissions policy](https://support.google.com/googleplay/android-developer/answer/14115180).
```
